site/src/xServices/workspaceSchedule/workspaceScheduleXService.ts

```diff
@@ -50,160 +53,227 @@ export type WorkspaceScheduleEvent =
-            actions: ["assignWorkspace"],
+      states: {
+        idle: {
+          tags: "loading",
+        },
+        gettingWorkspace: {
+          entry: ["clearGetWorkspaceError", "clearContext"],
+          invoke: {
+            src: "getWorkspace",
+            id: "getWorkspace",
+            onDone: {
+              target: "gettingPermissions",
+              actions: ["assignWorkspace"],
+            },
+            onError: {
+              target: "error",
+              actions: ["assignGetWorkspaceError"],
+            },
           },
-          onError: {
-            target: "error",
-            actions: ["assignGetWorkspaceError"],
+          tags: "loading",
+        },
+        gettingPermissions: {
+          entry: "clearGetPermissionsError",
+          invoke: {
+            src: "checkPermissions",
+            id: "checkPermissions",
+            onDone: [
+              {
+                actions: ["assignPermissions"],
+                target: "gettingTemplate",
+              },
+            ],
+            onError: [
+              {
+                actions: "assignGetPermissionsError",
+                target: "error",
+              },
+            ],
           },
+          tags: "loading",
         },
-        tags: "loading",
-      },
-      gettingPermissions: {
-        entry: "clearGetPermissionsError",
-        invoke: {
-          src: "checkPermissions",
-          id: "checkPermissions",
-          onDone: [
-            {
-              actions: ["assignPermissions"],
+        gettingTemplate: {
+          entry: "clearGetTemplateError",
+          invoke: {
+            src: "getTemplate",
+            id: "getTemplate",
+            onDone: {
               target: "presentForm",
+              actions: ["assignTemplate"],
             },
-          ],
-          onError: [
-            {
-              actions: "assignGetPermissionsError",
+            onError: {
               target: "error",
+              actions: ["assignGetTemplateError"],
             },
-          ],
-        },
-      },
-      presentForm: {
-        on: {
-          SUBMIT_SCHEDULE: "submittingSchedule",
+          },
+          tags: "loading",
         },
-      },
-      submittingSchedule: {
-        invoke: {
-          src: "submitSchedule",
-          id: "submitSchedule",
-          onDone: {
-            target: "submitSuccess",
-            actions: "displaySuccess",
+        presentForm: {
+          on: {
+            SUBMIT_SCHEDULE: {
+              target: "submittingSchedule",
+              actions: "assignAutoStopChanged",
+            },
           },
-          onError: {
-            target: "presentForm",
-            actions: ["assignSubmissionError"],
+        },
+        submittingSchedule: {
+          invoke: {
+            src: "submitSchedule",
+            id: "submitSchedule",
+            onDone: [
+              {
+                cond: "autoStartChanged",
```

Comment:
This shouldn't be "autoStopChanged"?

Reply:
Wow, it effectively is but I named it wrong. Good catch!